Libs/MRML/Core/vtkMRMLScene.cxx

```diff
@@ -1137,7 +1137,7 @@ vtkMRMLNode*  vtkMRMLScene::AddNodeNoNotify(vtkMRMLNode *n)
       // their references.
       std::string newId(sn->GetID());
       std::string oldId(n->GetID() ? n->GetID() : sn->GetID());
-      if (oldId != newId)
+      if (oldId != newId && !oldId.empty())
```

Comment:
Can oldId be empty? (by default node->ID is initialized to NULL and not "")
Shouldn't we check for GetID()==NULL instead?